File: ios/JumpingViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(JumpingViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onJump, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onText, RCTDirectEventBlock)
RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node)

@end

File: ios/JumpingViewManager.swift
import Foundation

@objc(JumpingViewManager)
class JumpingViewManager: RCTViewManager {
  override func view() -> UIView! {
    return JumpingView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! SquatCounterView
      component.update()
    }
  }
}
